hal/avr32/sonar_i2cxl.c

```diff
@@ -96,13 +98,41 @@ void sonar_i2cxl_get_last_measure(sonar_i2cxl_t* sonar_i2cxl)
 	
 	if ( distance_m > sonar_i2cxl->data.min_distance && distance_m < sonar_i2cxl->data.max_distance )
 	{
+		dt = ((float)time_us - sonar_i2cxl->data.last_update)/1000000.0f;
+
+		if (sonar_i2cxl->data.healthy)
+		{
+			velocity = (distance_m - sonar_i2cxl->data.current_distance) / dt;
+			if (abs(velocity)>20.0f)
```

Comment:
Yes, this is a casting to limit the range of the velocity. 
I used 20m/s value as it was already done in the barometer case (bmp085.c). 

But we can remove this check, or change the value (then also in bmp085) if you believe it is more realistic. 

In any case, a 20m/s vertical velocity is quite high anyway